src/chi/javascript/components/mobile-navigation.js

```diff
@@ -0,0 +1,458 @@
+          }
+        } else {
+          this._activateSecondLevelMenuItem(activator, drawerMenuItem, drawer.getAttribute('id'));
+          this.hide();
+        }
+      }
+    }
+  }
+
+  hide() {
+    const firstLevelDrawer = this._firstLevelDrawer;
+    const secondLevelDrawer = this._drawers.find(drawer => Util.hasClass(drawer, chi.classes.ACTIVE));
+
+    if (firstLevelDrawer) {
+      firstLevelDrawer.hide();
+    }
+    if (secondLevelDrawer) {
+      const firstLevelDrawerContent = this._mobileNavElement.querySelector(`.${FIRST_LEVEL_DRAWER_CONTENT_CLASS}`);
+
+      setTimeout(
+        function() {
+          Util.removeClass(secondLevelDrawer, chi.classes.ACTIVE);
+          Util.removeClass(firstLevelDrawerContent, SECOND_LEVEL_SHOWN_CLASS);
+        },
+        500
+      );
+    }
+  }
+
+  drawerMenuItemRemoveUnselected() {
+    const drawerActiveMenuItem = this._getActiveSecondLevelItem();
+
+    if (drawerActiveMenuItem && Util.checkHasClass(drawerActiveMenuItem, chi.classes.UNSELECTED)) {
+      Util.checkRemoveClass(drawerActiveMenuItem, chi.classes.UNSELECTED);
+    }
+  }
+
+  _toggleDrawerItemList(drawerMenuItem) {
+    const activeDrawerMenuItem = this._mobileNavElement.querySelector(`.${MOBILENAV_DRAWERS_CLASS} .${DRAWER_CLASS} .${DRAWER_CONTENT_CLASS} .${DRAWER_NAV_LIST} li.${chi.classes.ACTIVE}`);
+    const drawerMenuItemList = drawerMenuItem.querySelector(`.${DRAWER_ITEM_LIST_CLASS}`);
+    const expandedItem = drawerMenuItem.parentNode.querySelector(`.${DRAWER_ITEM_LIST_EXPANDED}`);
+    let drawerMenuItemListHeight;
+
+    if (this._menuItemAnimation) {
+      Util.stopThreeStepsAnimation(this._menuItemAnimation, false);
+    }
+
+    if (expandedItem && expandedItem !== drawerMenuItem) {
+      Util.removeClass(expandedItem, DRAWER_ITEM_LIST_EXPANDED);
+      expandedItem.querySelector(`.${DRAWER_ITEM_LIST_CLASS}`).style.removeProperty('height');
+    }
+
+    if (drawerMenuItem !== activeDrawerMenuItem &&
+      !Util.hasClass(drawerMenuItem, chi.classes.EXPANDED)) {
+      Util.checkAddClass(activeDrawerMenuItem, chi.classes.UNSELECTED);
+    } else {
+      this.drawerMenuItemRemoveUnselected();
+    }
+
+    function calculateHeight() {
+      drawerMenuItemList.style.position = 'absolute';
+      drawerMenuItemList.style.visibility = 'hidden';
+      drawerMenuItemList.style.display = 'block';
+      drawerMenuItemListHeight = window.getComputedStyle(drawerMenuItemList).height;
+      drawerMenuItemList.style.removeProperty('display');
+      drawerMenuItemList.style.removeProperty('visibility');
+      drawerMenuItemList.style.removeProperty('position');
+    }
+
+    if (!Util.checkHasClass(drawerMenuItem, DRAWER_ITEM_LIST_EXPANDED)) {
+      this._menuItemAnimation = Util.threeStepsAnimation(
+        function() {
+          calculateHeight();
+          Util.checkAddClass(drawerMenuItemList, chi.classes.TRANSITIONING);
+          drawerMenuItemList.style.height = '0px';
```

Comment:
I think units are not necessary when setting 0.
```suggestion
          drawerMenuItemList.style.height = '0';
```

Reply:
Looks like `px` must be present:
https://www.w3schools.com/Jsref/tryit.asp?filename=tryjsref_style_height

Reply:
It depends. It is not necessary for 0. The best way to handle this is to configure a linter.